concent_api/common/decorators.py

```diff
@@ -85,3 +89,32 @@ def decorated(*args: Any, **kwargs: Any) -> None:
                 return func(*args, **kwargs)
 
         return decorated
+
+
+class retry_on_database_error(object):
+
+    def __init__(self, func: Callable) -> None:
+        self.func = func
+        self.max_number_of_retries = settings.MAXIMUM_NUMBER_OF_RETRIES  # pylint: disable=attribute-defined-outside-init
+
+    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+        self._initialize_retry_parameters()
+        while True:
+            try:
+                result = self.func(*args, **kwargs)
+                break
+            except IntegrityError:
+                self._validate_number_of_retries()
+        return result
+
+    def _initialize_retry_parameters(self) -> None:
+        self.wait_time_in_sec = settings.STARTING_WAIT_TIME_FOR_RETRY_IN_SECONDS  # pylint: disable=attribute-defined-outside-init
+        self.current_number_of_retries = 0  # pylint: disable=attribute-defined-outside-init
+
+    def _validate_number_of_retries(self) -> None:
```

Comment:
Please rename to `_increase_retry_parameters`. There is no real validation here (data is not from client).